homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -272,16 +251,16 @@ def __init__(self, name, device, entry, unique_id, coordinator):
         """Initialize the generic Xiaomi device."""
         super().__init__(name, device, entry, unique_id, coordinator)
 
-        self._available = False
         self._available_attributes = {}
         self._state = None
         self._mode = None
         self._fan_level = None
-        self._state_attrs = {ATTR_MODEL: self._model}
+        self._state_attrs = {}
         self._device_features = 0
         self._supported_features = 0
         self._speed_count = 100
         self._preset_modes = []
+        self._operation_mode_class = Enum
```

Comment:
Maybe make this an abstractmethod property to ensure that child classes overrides this as it's required?

Reply:
`XiaomiGenericFan` and its child classes don't use `_operation_mode_class`.

Reply:
Then our distribution of features is probably not correct across the inheritance tree. Each feature should be scoped to where it's required at the most central location, but not further up the tree than that.

Reply:
I added `XiaomiGenericAirPurifier` class with `operation_mode_class` as abstractmethod property.